react/features/speaker-stats/components/web/SpeakerStatsButton.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+
+import { createToolbarEvent, sendAnalytics } from '../../../analytics';
+import { openDialog } from '../../../base/dialog';
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import AbstractSpeakerStatsButton from '../AbstractSpeakerStatsButton';
+
+import { SpeakerStats } from './';
+
+/**
+ * Implementation of a button for opening speaker stats dialog.
+ */
+class SpeakerStatsButton extends AbstractSpeakerStatsButton {
+
+    /**
+     * Handles clicking / pressing the button, and opens the appropriate dialog.
+     *
+     * @protected
+     * @returns {void}
+     */
+    _handleClick() {
+        const { dispatch, handleClick } = this.props;
+
+        if (handleClick) {
+            handleClick();
+
+            return;
+        }
+
+        sendAnalytics(createToolbarEvent('speaker.stats'));
```

Comment:
shouldn't we send analytics regardless of `handleClick` being truthy or not?

Reply:
i see you're sending it anyway in the `handleClick` so maybe remove it from there and just send it from here

Reply:
nvm, i confused things here